contracts/ERC721MLite.sol

```diff
@@ -0,0 +1,568 @@
+//SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+import "@openzeppelin/contracts/utils/cryptography/SignatureChecker.sol";
+import "erc721a/contracts/extensions/ERC721AQueryable.sol";
+import "./IERC721M.sol";
+
+/**
+ * @title ERC721MLite
+ *
+ * @dev ERC721A subclass with MagicEden launchpad features including
+ *  - multiple minting stages with time-based auto stage switch
+ *  - global and stage wallet-level minting limit
+ *  - whitelist using merkle tree
+ *  - crossmint support
+ *  - anti-botting
+ */
+contract ERC721MLite is IERC721M, ERC721AQueryable, Ownable, ReentrancyGuard {
```

Comment:
We will also need to have the Operator Filterer on this

Reply:
Unfortunately we cannot add it now. It's over size: 
<img width="1109" alt="image" src="https://github.com/magicoss/erc721m/assets/95395274/5fc503d1-1d54-474e-9041-2263e14887d8">

I will need to cut more things to make it under 24k

Reply:
Plan to bundle this with further down sizing effort.